server/ServerFSM.cpp

```diff
@@ -2166,6 +2166,24 @@ sc::result WaitingForTurnEndIdle::react(const SaveGameRequest& msg) {
     return transit<WaitingForSaveData>();
 }
 
+sc::result WaitingForTurnEndIdle::react(const HostlessSave& e) {
```

Comment:
Is a separate event needed for this? Could you just check in `react(const SaveGameRequest& msg)` if the server is in hostless mode, and if so, don't check for a sender connection or if its player is the host?

Reply:
I've tried but `SaveGameRequest` requires `PlayerConnectionPtr& player_connection` which cann't be `nullptr`.

Reply:
Yes. I suppose if `SaveGameRequest` doesn't have a connection then it was send in hostless mode.